### ISSUE TITLE
Advice on hyperparameters(loss,metrics) for multilabel y-hot with sparse triggers

### ISSUE THREAD

**Issue by JSunRae** [keras-team/keras#13920]
Hi all,

I have built a CNN to determine when a certain stock trades exist, and I'm getting muddled up trying to choose some hyperparameters.

The output data:
   -  has 6 labels of 0 or 1.
   -   There can be multiple correct labels ie [0, 1, 1, 0, 0, 0].
   -   In a batch of 22,000 there might only be 100 or 200 with a 1 label. So the true labels are sparse, and there are lots of zeros. So I'm trying to avoid the model just predicting zeros.
  -   Not sure if it helps, but the first three of the output at the long side, and the last three are the short side. So if there is a 1 in the first three labels, there should not be a 1 in the last 3 labels. ie:
      -   [ 1, 0, 0, 0, 0, 0] could exist
      -   [ 0, 0, 0, 1, 1, 1] could exist
      -   [ 1, 0, 1, 0, 0, 0] could exist
      -   [ 1, 0, 0, 1, 0, 0] would not exist
      -   [ 1, 0, 0, 0, 1, 0] would not exist

Here are my hyperparameters I have settled with:
Optimizer is 'nadam'
Last layer of CNN:
`layers.Dense(6, activation='sigmoid')`
Loss:
`loss=keras.losses.BinaryCrossentropy(from_logits=True)`
Metrics to monitor:
`metrics=[keras.losses.BinaryCrossentropy(from_logits=True, name='binary_crossentropy'), 'categorical_accuracy'`
Early Stopping in my callback:
`keras.callbacks.EarlyStopping(monitor='binary_crossentropy', patience=5)`

I'm now thinking I should be using a loss of weighted_crossentropy_with_logits, but really do not understand it and not sure.

**Comment by JSunRae** [comment]
I have not been able to work out how to calculate precision either.

**Comment by Saduf2019** [comment]
@JSunRae 
Moving this issue to closed status as there has been no activity, in case you still face the error please create a new issue.
